src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_load_store_emitters.cpp

```diff
@@ -65,18 +61,163 @@ void jit_load_emitter::emit_isa(const std::vector<size_t> &in_idxs, const std::v
+    }
+}
+
+template <cpu_isa_t isa>
+void jit_load_emitter::emit_isa(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs) const {
+    bool is_supported_precision = one_of(src_prc_, ov::element::f32, ov::element::i32, ov::element::f16, ov::element::i8, ov::element::u8) &&
+                                  (src_prc_ == dst_prc_ || one_of(dst_prc_, ov::element::f32, ov::element::i32));
+    OV_CPU_JIT_EMITTER_ASSERT(is_supported_precision, "Unsupported precision pair.");
+    OV_CPU_JIT_EMITTER_ASSERT(load_num_ <= static_cast<int>((get_vec_length() / dst_prc_.size())),
+                              "Unexpected number of elements to load.");
+
+    switch (src_prc_) {
+        case ov::element::f32:
+            load_qbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    break;
+                case ov::element::i32:
+                    cvt_f32_to_i32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                default:
+                    OV_CPU_JIT_EMITTER_THROW("Unsupported output type: ", dst_prc_.get_type_name());
+            }
+            break;
+        case ov::element::i32:
+            load_qbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    cvt_i32_to_f32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                case ov::element::i32:
+                    break;
+                default:
+                    OV_CPU_JIT_EMITTER_THROW("Unsupported output type: ", dst_prc_.get_type_name());
+            }
+            break;
+        case ov::element::f16:
+            load_dbyte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    cvt_f16_to_f32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                case ov::element::i32:
+                    cvt_f16_to_f32<isa>(h, aux_vec_idxs, aux_vec_idxs);
+                    cvt_f32_to_i32<isa>(h, aux_vec_idxs, out_idxs);
+                    break;
+                case ov::element::f16:
+                    break;
+                default:
+                    OV_CPU_JIT_EMITTER_THROW("Unsupported output type: ", dst_prc_.get_type_name());
+            }
+            break;
+        case ov::element::i8:
+        case ov::element::u8:
+            load_byte<isa>(in_idxs, src_prc_ == dst_prc_ ? out_idxs : aux_vec_idxs);
+            switch (dst_prc_) {
+                case ov::element::f32:
+                    cvt_byte_to_dbyte<isa>(h, aux_vec_idxs, aux_vec_idxs, src_prc_.is_signed());
+                    cvt_dbyte_to_i32<isa>(h, aux_vec_idxs, aux_vec_idxs, src_prc_.is_signed());
+                    cvt_i32_to_f32<isa>(h, aux_vec_idxs, out_idxs);
```

Comment:
If I got you, you specially declare all these `cvt` helpers in the separate file to call them in `convert` and `load/store` emitters.
But can only `convert` class have these all methods and `load/store` will just call `convert->emit_code()` in needed place?
What do you think?
From my perspective, if we have `jit_convert_emitter`, just let him be responsible for precision conversion. If another emitter needs the same code - just emit code of `jit_convert_emitter` with needed parameters.

What do you think about that? Probably I missed something here 🤔

Reply:
Good point! Applied. Thanks Alexandra!